apps/contribution/test/contribution.js

```diff
@@ -0,0 +1,231 @@
+      assert.equal(mintTokenPermission, true);
+    });
+  });
+
+  describe("Add contribution", async () => {
+    let amount = 100;
+    let contributorId = 1;
+    let hashDigest = '0x0000000000000000000000000000000000000000000000000000000000000000';
+    let hashFunction = 1;
+    let hashSize = 1;
+
+    it("should revert when add contribution from address that does not have permission", async () => {
+      return assertRevert(async () => {
+        await contribution.add(amount, contributorId, hashDigest, hashFunction, hashSize, {from: member1});
+        'sender does not have permission';
+      });
+    });
+
+    it("should revert when add contribution with amount equal to zero", async () => {
+      return assertRevert(async () => {
+        await contribution.add(0, contributorId, hashDigest, hashFunction, hashSize, {from: root});
+        'amount equal to zero';
+      });
+    });
+
+    it("add contribution", async () => {
+      let contributionCountBefore = await contribution.contributionsCount();
+      await contribution.add(amount, contributorId, hashDigest, hashFunction, hashSize, {from: root});
+      let contributionCountAfter = await contribution.contributionsCount();
+      // eslint-disable-next-line no-undef
+      assert.equal(contributionCountAfter.toNumber()-contributionCountBefore.toNumber(), 1);
+      let contributionObject = await contribution.getContribution(contributionCountAfter.toNumber());
+      // eslint-disable-next-line no-undef
+      assert.equal(contributionObject[1], contributorId);
+      let isExist = await contribution.exists(contributionCountAfter.toNumber());
+      // eslint-disable-next-line no-undef
+      assert.equal(isExist, true);
+    });
+  });
+
+  describe("Veto/Claim contribution", async () => {
+
+    it("should revert when veto from address that does not have permission", async () => {
+      let contributionId = await contribution.contributionsCount();
+      return assertRevert(async () => {
+        await contribution.veto(contributionId.toNumber(), {from: member1});
+        'sender does not have permission to veto';
+      });
+    });
+
+    it("should revert when veto contribution that does not exist", async () => {
+      let contributionId = await contribution.contributionsCount();
+      return assertRevert(async () => {
+        await contribution.veto(contributionId.toNumber()+1, {from: root});
+        'contribution not found';
+      });
+    });
+
+    it("should revert when claim contribution that does not exist", async () => {
+      let contributionId = await contribution.contributionsCount();
+      return assertRevert(async () => {
+        await contribution.claim(contributionId.toNumber()+1, {from: root});
+        'contribution not found';
+      });
+    });
+
+    it("claim contribution", async () => {
+      let contributionId = await contribution.contributionsCount();
+      await contribution.claim(contributionId);
```

Comment:
this one revert for some reasons

Reply:
Reverts unexpectedly? And in which context/scenario?

Reply:
in the scenario of claiming an added contribution

Reply:
How would that work? You can only claim confirmed contributions.

Reply:
ahhhh yup okay, missed this ` require(block.number >= c.confirmedAtBlock, 'NOT_CLAIMABLE'); ` thanks!